### DISCUSSION TITLE
Plugin / PanelMixin / Javascript file

### DISCUSSION THREAD

**Discussion by Bbillyben** [inventree/InvenTree#3937] [Q&A] [answered]
Hi there, 
I'm trying to take advantage of StockItemTracking and StockItemTestResult models by a plugin to display some tables in stock view or in the index.

I've inherited the plugin of `PanelMixin`, `SettingsMixin`, and `InvenTreePlugin`

I'm able to render a template ( => this would need a few words in the doc (folder structure and call) how can I add them? [Edit -> ok that's done] ) thanks to the `PluginTemplateLoader`

But I haven't found any clues of how to load a 'plugin javascript file', some sort of 'pluginstatic loader'

I've tried to add a script tag in the template, but as the file is not in the static path, it can't find it. 

Is there anything I missed? or any tags ? or anything to set up to be able to load/serve and execute som js script ?

[Edit] 

Is there any way to add a panel to the Index Page to add some sort of "dashboard" panel?

**Comment by wolflu05** [comment, marked as answer]
Hello, I was at the same point as you a while ago. The documentation is not very well documented for these integration plugins, but you can take a look at the doc strings in the code at: https://github.com/inventree/InvenTree/blob/master/InvenTree/plugin/base/integration/mixins.py#L610-L652 . Look at the properties you can append to your pannel object. There you'll find something for loading your script.

**Reply by SchrodingersGat** [reply]
@Bbillyben does the response by @wolflu05 help you here? Perhaps you would be able to submit some additional documentation if you feel it is not quite adequate.

**Reply by Bbillyben** [reply]
Many Thanks @wolflu05 !
The solution seems being that  'javascript_template' parameters of the panel, 
I'm trying to add the ref of my js file (stored in the template folder)

I've got a 302 error, but I'm able to log in console.

@SchrodingersGat  I will do some tests before marking as answer.

The doc state _clearly_ that many info are in the doc string, but I would pr some more info to point that parameter in the doc for the -un-reader like me!

**Reply by Bbillyben** [reply]
everything seems working like a charm, without the 302 error.... many thanks !
